C_CPP/Data Structures/Array/Matrix_multiplication/Matrix_multiplication.md

```diff
@@ -0,0 +1,145 @@
+
+## Program in c
+
+```C
+#include <stdio.h>
+
+// Driver function
+int main()
+{
+    int i, j, r1, c1, r2, c2, k; //Declaring variables
+
+    // taking inputs for number of rows and columns of two matrices
+    printf("Enter no. of rows of 1st matrix: ");
+    scanf("%d", &r1);
+    printf("Enter no. of columns of 1st matrix: ");
+    scanf("%d", &c1);
+    printf("Enter no. of rows of 2nd matrix: ");
+    scanf("%d", &r2);
+    printf("Enter no. of column of 2nd matrix: ");
+    scanf("%d", &c2);
+
+    // conditional statments
+    if (c1 != r2) //if number of columns of 1st matrix is not equal to no. of rows of 2nd matrix
+    {
+        printf("\nIt is not possible to multiply these two matrices,\n");
+        printf("because to multiply two matrices, \n");
+        printf("column of first matrix should be equal to \n");
+        printf("row of second matrix.");
+    }
+    else // condition if above condition does not satisfy
+    {
+        int m1[r1][c1], m2[r2][c2], m3[r1][c2]; //declaring varibles for matrices
+
+        printf("Enter matrix 1: \n");
+
+        for (i = 0; i < r1; i++) //loop to take input for elements of 1st matrix
+        {
+            for (j = 0; j < c1; j++)
+                scanf("%d", &m1[i][j]);
+        }
+
+        printf("Enter matrix 2: \n");
+
+        for (i = 0; i < r2; i++) //loop to take input for elements of 2nd matrix
+        {
+            for (j = 0; j < c2; j++)
+                scanf("%d", &m2[i][j]);
+        }
+
+        printf("Multplication of m1 and m2 is: \n");
+
+        //loop to multiply matrices m1 and m2
+        for (i = 0; i < r1; i++)
+        {
+            for (j = 0; j < c2; j++)
+            {
+                int sum = 0;
+                for (k = 0; k < c1; k++)
+                {
+                    //multiplying two matrices and then adding with sum variable
+                    sum += m1[i][k] * m2[k][j];
+                }
+                m3[i][j] = sum; //storing final result of matrix multiplication in m3
+                printf("%d ", m3[i][j]);
+            }
+            printf("\n");
+        }
+    }
+    return 0;
+}
+```
+
+## Sample tests
+
+1. if column of 1<sup>st</sup> matrix is not equal to row of 2<sup>nd</sup> matrix
+
+```
+Enter no. of rows of 1st matrix: 2
+Enter no. of columns of 1st matrix: 2
+Enter no. of rows of 2nd matrix: 3
+Enter no. of column of 2nd matrix: 2
+
+It is not possible to multiply these two matrices,
+because to multiply two matrices,
+column of first matrix should be equal to
+row of second matrix.
+```
+
+2. if column of 1<sup>st</sup> matrix and row of 2<sup>nd</sup> matrix are equal
+
+```
+Enter no. of rows of 1st matrix: 2
+Enter no. of columns of 1st matrix: 2
+Enter no. of rows of 2nd matrix: 2
+Enter no. of column of 2nd matrix: 2
+Enter matrix 1: 
+1 2
+3 4
+Enter matrix 2: 
+5 6
+7 8
+Multplication of m1 and m2 is: 
+19 22
+43 50
+```
+
+
+# References
+
+https://en.wikipedia.org/wiki/Matrix_multiplication
```

Comment:
add links like this [Google](www.google.com)
![website](website link)

Reply:
Changes done, please review it